### DISCUSSION TITLE
VScode commands for Langium extensions

### DISCUSSION THREAD

**Discussion by lazkany** [eclipse-langium/langium#1235] [Q&A] [answered]
Hi, I have followed the tutorials and everything worked for me. I have now a language with a generator to java that works perfectly.
Now I am trying to register a vscode command so that I read the current file in the editor and directly trigger the generator. This way is easier than manually trigger the cli.  The way I understand is that such commands are registered inside the active function in the file extension.ts.

However, it seems that Languim disabled this and if I write a silly command just like the one below, it does not show up. Am I missing something?

Thanks

```
export function activate(context: vscode.ExtensionContext): void {
    client = startLanguageClient(context);
    let disposable= vscode.commands.registerCommand('vistodo.hello', ()=>{
        vscode.window.showInformationMessage('Welcome to SynTM');
    });
    context.subscriptions.push(disposable);
}
```

**Comment by msujew** [comment, marked as answer]
Hey @lazkany,

You will need to [declare the command](https://code.visualstudio.com/api/references/contribution-points#contributes.commands) in the package.json of your extension. While the command that you created is still registered in vscode, it isn't available in the command palette without the declaration.

**Reply by lazkany** [reply]
That is perfect. Thanks a lot!

I really enjoy working with Languim and the full support.

**Reply by msujew** [reply]
You're welcome. Glad to hear you enjoy working with Langium :)
